linkerd/proxy/balance/src/discover/buffer.rs

```diff
@@ -16,48 +16,77 @@ where
 {
     let (tx, rx) = mpsc::channel(capacity);
 
+    let send = |tx: &mpsc::Sender<_>, change| {
+        if let Err(e) = tx.try_send(change) {
+            match e {
+                // The balancer has been dropped (and will never be used again).
+                mpsc::error::TrySendError::Closed(_) => {
+                    debug!("Discovery receiver dropped");
+                }
+
+                // The balancer is stalled and we can't continue to buffer
+                // updates for it.
+                mpsc::error::TrySendError::Full(_) => {
+                    warn!("The balancer is not processing discovery updates; aborting discovery stream");
+                }
+            }
+            return false;
+        }
+        true
```

Comment:
nit, take it or leave it: why not just 
```suggestion
        match tx.try_send(change) {
            // The balancer has been dropped (and will never be used again).
            Err(mpsc::error::TrySendError::Closed(_)) => {
                debug!("Discovery receiver dropped");
                false
            },

            // The balancer is stalled and we can't continue to buffer
            // updates for it.
            Err(mpsc::error::TrySendError::Full(_)) => {
                warn!("The balancer is not processing discovery updates; aborting discovery stream");
                false
            },

            Ok(()) => true,
        }
```
which imo is a little easier to follow